### ISSUE TITLE
If not false, then...

### ISSUE THREAD

**Issue by franz-josef-kaiser** [tollmanz/ostrichcize#5]
[Here](https://github.com/tollmanz/ostrichcize/blob/master/ostrichcize.php#L99), you can simply ask

```
apply_filters( 'ostrichcize_theme', false ) AND $this->_paths[] = get_stylesheet_directory();
```

...and [here](https://github.com/tollmanz/ostrichcize/blob/master/ostrichcize.php#L148) 

```
// either - or
if ( strpos( $errstr, $path ) XOR strpos( $errfile, $path ) )
    return true;
```

**Comment by tollmanz** [comment]
I'm afraid that your brain is able to parse these things much better than mine ;) I think these are much more clever implementations than I currently use (they certainly require less code), but the trade off is poor readability. I find the suggestions tougher to comprehend, which potentially makes the code harder for others to step into.

What would be the justification for making the change?

**Comment by franz-josef-kaiser** [comment]
My brain certainly has its limits too. It's melting since I found the following working :)

<pre>
public function __construct( $foo = null )
{
    $this->foo = $foo ?: $this->bar;
}
</pre>

About the other examples above:

When I was starting with PHP, I found it extremely hard to understand the following line

> IF NOT FALSE IS result-from-function RETURN BOOL

so I switched to doing simpler to understand one liners

> IF TRUE AND/THEN foo

About using `XOR`) It's simply the selector that should be used in such cases. It's also faster than questioning something two times.
About not using `false !==`) I think it's much easier to understand if the expected return value is met, than understanding if the return value is met _and_ `false NOT IS`.
About readability) If it needs to be formatted for readability, then I'd go with an approach like this:

<pre>
if ( 
    strpos( $errstr, $path )
    XOR
    strpos( $errstr, $path )
)
    return true;
</pre>
